src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -653,7 +654,15 @@ export class RunActiveFileInTerminalAction extends Action {
 			this.notificationService.warn(nls.localize('workbench.action.terminal.runActiveFile.noFile', 'Only files on disk can be run in the terminal'));
 			return TPromise.as(void 0);
 		}
-		instance.sendText(uri.fsPath, true);
+		let uriPath: string = uri.fsPath;
+		if ((uriPath.indexOf(' ') !== -1) || (uriPath.indexOf('"') !== -1)) {
+			if (isWindows) {
+				uriPath = '"' + uriPath + '"';
+			} else {
+				uriPath = '\'' + uriPath + '\'';
```

Comment:
Escaping on macOS and Linux would probably be better done by escaping each space, this is more typical than quoting the whole path. This is what happens when you drag a file in that contains spaces: `/Users/daimms/Desktop/Screen\ Shot\ 2018-09-14\ at\ 12.39.14\ PM.png`